include/seqan3/alphabet/nucleotide/nucleotide_base.hpp

```diff
@@ -47,17 +47,26 @@ namespace seqan3
 {
 
 /*!\brief A CRTP-base that refines seqan3::alphabet_base and is used by the nucleotides.
+ * \ingroup nucleotide
  * \tparam derived_type The CRTP parameter type.
  * \tparam size         The size of the alphabet.
- * \tparam char_t       The character type of the alphabet (set this to `void` when defining just a
- *                      seqan3::semi_alphabet_concept).
+ *
+ * \details
+ *
+ * You can use this class to define your own nucleotide alphabet, but types are not required to be based on it to model
+ * seqan3::nucleotide_concept, it is purely a way to avoid code duplication.
+ *
+ * The derived type needs to define only the following table as static member variable:
+ *
+ *   * `static std::array<THAT_TYPE, value_size> complement_table` that defines for every possible rank value
```

Comment:
`THAT_TYPE` is not further described, also somehow the other tables `rank_to_char` and `char_to_rank` are missing.

Reply:
> THAT_TYPE is not further described,

How would you change it` `derived_type`?

> also somehow the other tables rank_to_char and char_to_rank are missing.

They are provided by nucleotide_base to the derived class does not need to offer them.

Reply:
> > THAT_TYPE is not further described,
> 
> How would you change it `derived_type`?

Maybe like this:

```c++
// * The derived type (i.e. THAT_TYPE) needs to define only the following table as static member variable:
```

or 

```c++
 * The derived type needs to define only the following table as static member variable:
 *
 *   * `static std::array<THAT_TYPE, value_size> complement_table` that defines for every possible rank value
 *     the corresponding complement.
 * 
 * where THAT_TYPE is the name of the derived type which inherits seqan3::nucleotide_base (e.g. seqan3::dna4, ...)
```

> 
> > also somehow the other tables rank_to_char and char_to_rank are missing.
> 
> They are provided by nucleotide_base to the derived class does not need to offer them.

? dna4, dna5,... still provide rank_to_char and char_to_rank and are not provided by nucleotide_base. Maybe I understand your answer wrong, but I think we should mention that nucleotide_base inherits the requirements of alphabet_base ans must be provided by the derived_type.